addons/animated-thumb/userscript.js

```diff
@@ -148,22 +147,26 @@ export default async function ({ addon, console, msg }) {
         .finally(() => {
           ignoreClickOutside = false;
           uploadFromFileButton.removeAttribute("disabled");
+          uploadFromFileButton.classList.remove("loading");
           uploadFromStageButton.removeAttribute("disabled");
+          uploadFromStageButton.classList.remove("loading");
         });
 
     const upload = () => {
-      modalResultArea.className = "sa-animated-thumb-popup-result sa-animated-thumb-popup-result-none";
+      modalResultArea.classList.add("hidden");
       uploadFromFileButton.setAttribute("disabled", "true");
       uploadFromStageButton.setAttribute("disabled", "true");
     };
 
-    uploadFromFileButton.addEventListener("click", () => {
+    uploadFromFileButton.addEventListener("click", (e) => {
+      e.target.classList.add("loading");
```

Comment:
```suggestion
      uploadFromFileButton.classList.add("loading");
```
The target could theoretically be a child of the button - there aren't any right now, but the UI could change in the future.